include/pybind11/cast.h

```diff
@@ -2172,16 +2172,35 @@ class unpacking_collector {
     dict m_kwargs;
 };
 
+#if defined(__INTEL_COMPILER) && defined(PYBIND11_CPP17)
+// We need this Intel compiler workaround because it fails to compile
+// the enable_if_t<!all_of<is_positional<Args>...>::value> part below
+// (tested with ICC 2021.1 Beta 20200827).
+template <typename... Args>
+constexpr bool args_are_all_positional()
+{
+  return all_of<is_positional<Args>...>::value;
```

Comment:
Should we bite the bullet and use this unconditionally? The comment can still stay.

Reply:
Sounds reasonable, so I changed it. The two variants are absolutely identical.